korau/src/commonMain/kotlin/korlibs/audio/sound/backends/ALSA.kt

```diff
@@ -45,15 +46,17 @@ class ALSAPlatformAudioOutput(
     }
 
     override suspend fun add(samples: AudioSamples, offset: Int, size: Int) {
-        if (!ASound2.initialized) return super.add(samples, offset, size)
+        if (size == 0) return
+        if (!ASound2.initialized) return super.add(samples, offset, samples.totalSamples)
         //if (!running) start()
 
         if (!running) delay(10.milliseconds)
 
         while (running && lock { sdeque.availableRead > 4 * 1024 }) {
             delay(10.milliseconds)
         }
-        lock { sdeque.write(samples, offset, size) }
+        availableSamples += samples.totalSamples
+        lock { sdeque.write(samples, offset, samples.totalSamples) }
```

Comment:
might make sense to put `availableSamples +=` inside the lock to ensure it is thread safe?